jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js

```diff
@@ -0,0 +1,140 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('inspector-plans', function() {
+
+  juju.components.InspectorPlans = React.createClass({
+    propTypes: {
+      acl: React.PropTypes.object.isRequired,
+      changeState: React.PropTypes.func.isRequired,
+      listPlansForCharm: React.PropTypes.func.isRequired,
+      service: React.PropTypes.object.isRequired
+    },
+
+    plansXHR: null,
+
+    getInitialState: function() {
+      return {
+        plans: [],
+        plansLoading: false
+      };
+    },
+
+    componentDidMount: function() {
+      this._getPlans();
+    },
+
+    componentWillUnmount: function() {
+      if (this.plansXHR) {
+        this.plansXHR.abort();
+      }
+    },
+
+    /**
+      Get the list of plans available for the service.
+
+      @method _getPlans
+    */
+    _getPlans: function() {
+      this.setState({plansLoading: true}, () => {
+        this.plansXHR = this.props.listPlansForCharm(
```

Comment:
In theory the list of plans could be already present as an attribute of the charm. retrieved when the charm detail was loaded. So we could save a roundtrip in that case. But not for this branch, maybe add a TODO?